.github/workflows/asana.yml

```
@@ -29,9 +29,17 @@ on:
         required: false
         description: GitHub secret that Asana uses to fetch PR information.
 jobs:
+  check-for-secrets:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
```

I believe even with the `$GITHUB_OUTPUT` approach you'll need to define the outputs on the job like that:

```suggestion
    runs-on: ubuntu-latest
    outputs:
      has-asana-token: ${{ steps.check-inputs.outputs.has-asana-token }}
      has-github-secret: ${{ steps.check-inputs.outputs.has-github-secret }}
    steps:
      - id: check-inputs
      - run: |
```

(Checked against [this linter tool](https://rhysd.github.io/actionlint/))